src/common.js

```diff
@@ -17,7 +17,10 @@ export const checkStatus = (response) => {
 
   let error = new Error(response.statusText);
   error.response = response;
-  throw error;
+
+  return response.json()
+    .then(json => Promise.reject(Object.assign(error, json)))
```

Comment:
Ah, that's not bad either. I'm fine with either one, as long as the tests agree with it :)